ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -104,9 +108,33 @@ protected boolean removeEldestEntry(Map.Entry<String, Boolean> eldest) {
             return size() > cacheMaxLimit;
           }
         }));
-    this.writeCacheAfterUpdate = config.writeCacheAfterUpdate;
+
+    accountMigrationService =
+        new AccountMigrationService(backupFileManager, mySqlAccountStore, needRefresh, lastSyncTime,
+            recentNotFoundContainersCache, accountInfoMapRef, accountServiceMetrics.getAccountServiceMetrics(),
+            infoMapLock, config, notifier, scheduler);
+
+    if (config.enableNewDatabaseForMigration) {
+      backupFileManagerNew =
+          new BackupFileManager(accountServiceMetrics, config.backupDirNew, config.maxBackupFileCount);
+      recentNotFoundContainersCacheNew = Collections.newSetFromMap(
+          Collections.synchronizedMap(new LinkedHashMap<String, Boolean>(cacheInitialCapacity, cacheLoadFactor, true) {
+            protected boolean removeEldestEntry(Map.Entry<String, Boolean> eldest) {
+              return size() > cacheMaxLimit;
+            }
+          }));
+      accountInfoMapRefNew = new AtomicReference<>(new AccountInfoMap(accountServiceMetrics.getAccountServiceMetricsNew()));
+      accountMigrationServiceNew =
+          new AccountMigrationService(backupFileManagerNew, mySqlAccountStoreNew, needRefreshNew, lastSyncTimeNew,
+              recentNotFoundContainersCacheNew, accountInfoMapRefNew,
+              accountServiceMetrics.getAccountServiceMetricsNew(), infoMapLock, config, notifier, scheduler);
+    }
+
     // Initialize cache from backup file on disk
-    initCacheFromBackupFile();
+    accountMigrationService.initCacheFromBackupFile();
+    if (config.enableNewDatabaseForMigration) {
+      accountMigrationServiceNew.initCacheFromBackupFile();
+    }
     // Fetches added or modified accounts and containers from mysql db and schedules to execute it periodically
     initialFetchAndSchedule();
```

Comment:
this method can go to MigraitonService as well.

Reply:
updated.